plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2087,6 +2046,8 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
             process_scheduled_and_incoming_trxs( scheduled_trx_deadline, incoming_itr );
          }
 
+         process_read_only_transactions( preprocess_deadline );
```

Comment:
Should this be renamed something like `repost_exhausted_transactions`?

Reply:
Yes, and should honor the deadline.